pkcs11/src/new/types/mechanism/mod.rs

```diff
@@ -339,356 +363,397 @@ pub enum Mechanism {
+            MechanismType::AesEcbEncryptData => CKM_AES_ECB_ENCRYPT_DATA,
+            MechanismType::AesCbcEncryptData => CKM_AES_CBC_ENCRYPT_DATA,
+            MechanismType::Gostr3410KeyPairGen => CKM_GOSTR3410_KEY_PAIR_GEN,
+            MechanismType::Gostr3410 => CKM_GOSTR3410,
+            MechanismType::Gostr3410WithGostr3411 => CKM_GOSTR3410_WITH_GOSTR3411,
+            MechanismType::Gostr3410KeyWrap => CKM_GOSTR3410_KEY_WRAP,
+            MechanismType::Gostr3410Derive => CKM_GOSTR3410_DERIVE,
+            MechanismType::Gostr3411 => CKM_GOSTR3411,
+            MechanismType::Gostr3411Hmac => CKM_GOSTR3411_HMAC,
+            MechanismType::Gost28147KeyGen => CKM_GOST28147_KEY_GEN,
+            MechanismType::Gost28147Ecb => CKM_GOST28147_ECB,
+            MechanismType::Gost28147 => CKM_GOST28147,
+            MechanismType::Gost28147Mac => CKM_GOST28147_MAC,
+            MechanismType::Gost28147KeyWrap => CKM_GOST28147_KEY_WRAP,
+            MechanismType::DsaParameterGen => CKM_DSA_PARAMETER_GEN,
+            MechanismType::DhPkcsParameterGen => CKM_DH_PKCS_PARAMETER_GEN,
+            MechanismType::X9_42DhParameterGen => CKM_X9_42_DH_PARAMETER_GEN,
+            MechanismType::DsaProbabilisticParameterGen => CKM_DSA_PROBABLISTIC_PARAMETER_GEN,
+            MechanismType::DsaShaweTaylorParameterGen => CKM_DSA_SHAWE_TAYLOR_PARAMETER_GEN,
+            MechanismType::AesOfb => CKM_AES_OFB,
+            MechanismType::AesCfb64 => CKM_AES_CFB64,
+            MechanismType::AesCfb8 => CKM_AES_CFB8,
+            MechanismType::AesCfb128 => CKM_AES_CFB128,
+            MechanismType::AesCfb1 => CKM_AES_CFB1,
+            MechanismType::VendorDefined => CKM_VENDOR_DEFINED,
+            MechanismType::Sha224 => CKM_SHA224,
+            MechanismType::Sha224Hmac => CKM_SHA224_HMAC,
+            MechanismType::Sha224HmacGeneral => CKM_SHA224_HMAC_GENERAL,
+            MechanismType::Sha224RsaPkcs => CKM_SHA224_RSA_PKCS,
+            MechanismType::Sha224RsaPkcsPss => CKM_SHA224_RSA_PKCS_PSS,
+            MechanismType::Sha224KeyDerivation => CKM_SHA224_KEY_DERIVATION,
+            MechanismType::CamelliaEcb => CKM_CAMELLIA_ECB,
+            MechanismType::CamelliaCbc => CKM_CAMELLIA_CBC,
+            MechanismType::CamelliaMac => CKM_CAMELLIA_MAC,
+            MechanismType::CamelliaMacGeneral => CKM_CAMELLIA_MAC_GENERAL,
+            MechanismType::CamelliaCbcPad => CKM_CAMELLIA_CBC_PAD,
+            MechanismType::CamelliaEcbEncryptData => CKM_CAMELLIA_ECB_ENCRYPT_DATA,
+            MechanismType::CamelliaCbcEncryptData => CKM_CAMELLIA_CBC_ENCRYPT_DATA,
+            MechanismType::AesKeyWrap => CKM_AES_KEY_WRAP,
+            MechanismType::AesKeyWrapPad => CKM_AES_KEY_WRAP_PAD,
+            MechanismType::RsaPkcsTpm_1_1 => CKM_RSA_PKCS_TPM_1_1,
+            MechanismType::RsaPkcsOaepTpm_1_1 => CKM_RSA_PKCS_OAEP_TPM_1_1,
+            */
+        }
+    }
+}
+
+impl TryFrom<CK_MECHANISM_TYPE> for MechanismType {
```

Comment:
This is the only way to create new `MechanismType` from an integer value. It only creates good-formed values so I reckoned having a `panic` in the other direction was ok.

Reply:
You can probably still transmute memory to get a bad value, but shame on whoever does that.

Yeah, a panic in the other direction works.